### ISSUE TITLE
lstm time series forecasting python tensorflow

### ISSUE THREAD

**Issue by snake-plant-care-indoors** [tensorflow/tensorflow#83148]
1. **Import Libraries**:  
   ```python
   import numpy as np  
   import pandas as pd  
   from tensorflow.keras.models import Sequential  
   from tensorflow.keras.layers import LSTM, Dense  
   from sklearn.preprocessing import MinMaxScaler  
   ```

2. **Load Dataset**: Load your time series data using pandas.  
   ```python
   data = pd.read_csv('your_data.csv')  
   ```

3. **Preprocess Data**: Normalize the dataset using MinMaxScaler.  
   ```python
   scaler = MinMaxScaler()  
   data_scaled = scaler.fit_transform(data)  
   ```

4. **Create Sequences**: Split the data into sequences of input (X) and output (y).  
   ```python
   def create_sequences(data, time_steps=1):  
       X, y = [], []  
       for i in range(len(data) - time_steps):  
           X.append(data[i:i+time_steps])  
           y.append(data[i+time_steps])  
       return np.array(X), np.array(y)  
   time_steps = 10  
   X, y = create_sequences(data_scaled, time_steps)  
   ```

5. **Split Data**: Divide data into training and testing sets.  
   ```python
   train_size = int(len(X) * 0.8)  
   X_train, X_test = X[:train_size], X[train_size:]  
   y_train, y_test = y[:train_size], y[train_size:]  
   ```

6. **Reshape Input**: Reshape X data to fit LSTM input shape.  
   ```python
   X_train = X_train.reshape((X_train.shape[0], X_train.shape[1], 1))  
   X_test = X_test.reshape((X_test.shape[0], X_test.shape[1], 1))  
   ```

7. **Build LSTM Model**: Create an LSTM model.  
   ```python
   model = Sequential()  
   model.add(LSTM(50, activation='relu', input_shape=(time_steps, 1)))  
   model.add(Dense(1))  
   model.compile(optimizer='adam', loss='mse')  
   ```

8. **Train Model**: Fit the model on the training data.  
   ```python
   model.fit(X_train, y_train, epochs=50, batch_size=32, validation_data=(X_test, y_test))  
   ```

9. **Make Predictions**: Use the model to predict future values.  
   ```python
   predictions = model.predict(X_test)  
   predictions = scaler.inverse_transform(predictions)  
   ```

10. **Evaluate Results**: Compare predictions with actual values to check accuracy.  
   ```python
   from sklearn.metrics import mean_squared_error  
   mse = mean_squared_error(scaler.inverse_transform(y_test), predictions)  
   print(f'MSE: {mse}')  
   ```

**Comment by tilakrayal** [comment]
@snake-plant-care-indoors,
I tried to execute the mentioned code and observed that it was executed without any issues/error. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/1c3a09084db3e3a74ca39425bba89f0e/untitled2281.ipynb) and also have a look at this official document for the time series forecasting using tensorflow.

https://www.tensorflow.org/tutorials/structured_data/time_series

Thank you!
